libs/ledger-live-common/src/hw/installLanguage.ts

```diff
@@ -0,0 +1,189 @@
+import { Observable, from, of, throwError, EMPTY } from "rxjs";
+import { catchError, concatMap, delay, mergeMap } from "rxjs/operators";
+import { DeviceOnDashboardExpected, TransportError, TransportStatusError } from "@ledgerhq/errors";
+
+import ManagerAPI from "../api/Manager";
+import { withDevice } from "./deviceAccess";
+import getDeviceInfo from "./getDeviceInfo";
+import { Language, languageIds, LanguagePackage } from "../types/languages";
+import network from "../network";
+import { LanguageInstallRefusedOnDevice } from "../errors";
+import { AppAndVersion } from "./connectApp";
+import appSupportsQuitApp from "../appSupportsQuitApp";
+import quitApp from "./quitApp";
+import getAppAndVersion from "./getAppAndVersion";
+import { isDashboardName } from "./isDashboardName";
+import Transport from "@ledgerhq/hw-transport";
+
+export type InstallLanguageEvent =
+  | {
+      type: "appDetected";
+    }
+  | {
+      type: "unresponsiveDevice";
+    }
+  | {
+      type: "progress";
+      progress: number;
+    }
+  | {
+      type: "devicePermissionRequested";
+    }
+  | {
+      type: "languageInstalled";
+    };
+
+const attemptToQuitApp = (transport, appAndVersion?: AppAndVersion): Observable<InstallLanguageEvent> =>
+  appAndVersion && appSupportsQuitApp(appAndVersion)
+    ? from(quitApp(transport)).pipe(
+        concatMap(() =>
+          of(<InstallLanguageEvent>{
+            type: "unresponsiveDevice",
```

Comment:
Why on a successful quit (i think it's successful because we don't catch an error here) do we return an event of type `unresponsiveDevice` ? What does `unresponsiveDevice` means here ?

Reply:
This was inspired by the same function that exists on the connectManager commend. IIUC it's for retriggering the command attempts once the device has quit the app. @juan-cortes can you confirm that?

Reply:
Can we reuse the one from the manager?
And indeed, device actions restart when a device changes connection stack so emitting this acts as placeholder while we disconnect and reconnect the device as we exit the wrong app and land in the dashboard.

Reply:
I think we can. But I have to check what to do with the typing. As now it's specific to each of the actions. I'm on it.

Reply:
Done. But when testing it I found a bug, it's not retriggering the action for the language install after we quit and app (for manager it's still working). I think it's an issue with the useEffect within the action hook. I'll investigate further tomorrow

Reply:
Is this solved?

Reply:
Yes. it's solved.